### ISSUE TITLE
Need param in config to disable MinMaxScaler for FreqAI

### ISSUE THREAD

**Issue by ducky-duke** [freqtrade/freqtrade#10852]
[https://github.com/dmlc/xgboost/issues/357](https://github.com/dmlc/xgboost/issues/357)
[https://datascience.stackexchange.com/questions/60950/is-it-necessary-to-normalize-data-for-xgboost](https://datascience.stackexchange.com/questions/60950/is-it-necessary-to-normalize-data-for-xgboost)

I suggest there is a param to disable the MinMaxScaler for FreqAI. I see that the MinMaxScaler remove the details of the data.

Example range value of an indicator is from 0-100, but it only appear from 40-100 in my data, so apply MinMaxScaler at here is not good. It's only good when you know the full range value of that data.

**Comment by xmatthias** [comment]
Please get used to filling out the issue template.  Those who frequently omit the template may have their issues closed without comment.
We're asking some very basic questions - by deleting the issue template, you're explicitly choosing not to answer our questions.

Filling out the template helps reduce our workload, and we greatly appreciate that.

---


That "feature" exists already.
FreqAI Provides a [default data pipeline](https://www.freqtrade.io/en/stable/freqai-feature-engineering/#building-the-data-pipeline) - but you can customize that datapipeline and define your own by implementing [`define_data_pipeline()`](https://www.freqtrade.io/en/stable/freqai-feature-engineering/#customizing-the-pipeline) in your custom FreqAI model.

Now the builtin pipeline does a bit more (see below) - so unless you re-implement the parts you need (or all) - you'll lose some other functionalities. 
Best check through the code to see what exactly exists - and then determine if you're in need of this or not.

https://github.com/freqtrade/freqtrade/blob/930b5e07f811ebe302a139b2942e3efacf2059d2/freqtrade/freqai/freqai_interface.py#L523-L551

**Comment by ducky-duke** [comment]
@xmatthias yes, i already edit the pipeline, but i hope to have any param in config file, is it good to implement ?

**Comment by robcaulk** [comment]
> but it only appear from 40-100 in my data,

It seems that you are missing the entire point of data normalization. The range of your data actually does *not* matter, which is precisely the point of normalization. In taking 40-100 and converting it to -1 and 1, you improve your training accuracy and speed. If you dont do this, you risk unwanted weightings across different features, slow downs, and worse error plateaus (although it depends entirely on the library/model you are using, we offer 18....). Now it could be that the library you chose (I guess XGBoost but you never said explicitly) does not require it, but then you miss the point that subsequent steps in your data pipeline *do* require scaling! E.g. DI, SVM, Noise, and PCA!!

Dont forget, that the next time FreqAI retrains a model for you, if your data has shifted to 50-90, FreqAI will also build a new data pipeline for you, and FreqAI will inference that model with the new data pipeline, so again, normalization is handled perfectly for you.

But if you prefer to have less accurate training, and slower training, and risk ruining the rest of your pipeline dteps go right ahead - simply redefine the data pipeline in your FreqAI model with which ever steps you do or dont want:

https://www.freqtrade.io/en/stable/freqai-feature-engineering/#building-the-data-pipeline

But I don't recommend doing that unless you replace it with different scaler.

No, we won't make a config parameter, because generally you would be replacing it with a different normalization step of your choice (for example, the `QuantileTransformer` as shown in the example linked above, but it could be any step available in SKLearn, or even custom steps you created yourself).

> It's only good when you know the full range value of that data.

FreqAI is built on the idea that your training parameter space is fixed, you train a data pipeline for it, then you can apply the same transofmation to your inference. If your training data is 40-100, and your inference is 35, that is important information to know about because it helps your outlier detection see that the inference is not well trained on (again with DI, SVM etc). 

Then you retrain on new data, and you have a new fixed training parameter space.
